compiler/tests-full/stdlib.cma.expected.js

```diff
@@ -1292,7 +1273,7 @@
      /*<<obj.ml:107:4>>*/ return runtime.caml_ephe_create(l) /*<<obj.ml:107:12>>*/ ;
    }
    function length(x){
-     /*<<obj.ml:109:17>>*/ return x.length - 3 | 0;
+     /*<<obj.ml:109:17>>*/ return x.length - 1 - 2 | 0;
```

Comment:
I'm wondering why this is no longer optimized.

Reply:
This was  previously optimized by 
```ocaml
  register_bin_prims
    [ "%int_add"; "caml_int32_add"; "caml_nativeint_add" ]
    `Pure
    (fun cx cy _ ->
      match cx, cy with
      | J.EBin (J.Minus, cz, J.ENum n), J.ENum m ->
          to_int (J.EBin (J.Plus, cz, J.ENum (J.Num.add m (J.Num.neg n))))
      | _ -> to_int (plus_int cx cy));
  register_bin_prims
    [ "%int_sub"; "caml_int32_sub"; "caml_nativeint_sub" ]
    `Pure
    (fun cx cy _ ->
      match cx, cy with
      | J.EBin (J.Minus, cz, J.ENum n), J.ENum m ->
          to_int (J.EBin (J.Minus, cz, J.ENum (J.Num.add n m)))
      | _ -> to_int (J.EBin (J.Minus, cx, cy)));
```